contracts/BitmaskRBAC.sol

```diff
@@ -87,30 +98,41 @@ contract BitmaskRBAC is RBAC {
     return userCountsByRole[_role];
   }
 
+  function checkRole(address _operator, string _role) public view returns (bool) {
+    uint256 existingRoleBitmask = users[_operator].roleBitmask;
+    uint position = rolePosition[_role];
+    return existingRoleBitmask.checkBit(position);
+  }
+
   function addUserRole(string _role)
   onlyRbacAdmin
   external {
-    require(supportedRoleList.length < 256); // because we use a uint256 as a bitmask
+    uint numRoles = supportedRoleList.length;
+    require(numRoles < 256); // because we use a uint256 as a bitmask
     require((bytes)(_role).length > 0);
     require(!roleExists(_role));
+
     supportedRoles[_role] = true;
     supportedRoleList.push(_role);
+    rolePosition[_role] = numRoles;
   }
 
-  function _grantRoleIfNotGrantedNoChecks(address user, string roleName)
-  internal {
+  function grantRole(address user, string roleName)
+  onlyRbacAdmin
+  checkUserExists(user)
+  checkRoleExists(roleName)
+  public {
     if (!hasRole(user, roleName)) {
       userCountsByRole[roleName]++;
       addRole(user, roleName);
     }
   }
 
-  function grantRole(address user, string roleName)
-  onlyRbacAdmin
-  checkUserExists(user)
-  checkRoleExists(roleName)
-  public {
-    _grantRoleIfNotGrantedNoChecks(user, roleName);
+  function addRole(address _operator, string _roleName) internal {
```

Comment:
Unsafe to call this without first checking that the role exists. Given that fact, and the fact that we only call this once anyway, maybe this function should disappear and its code get folded into grantRole?